src/dataflow/src/render/reduce.rs

```diff
@@ -714,14 +1127,13 @@ where
     };
 
     let mut to_aggregate = Vec::new();
-    let diffs_len = aggrs.len() * 3;
+    let diffs_len = num_accumulable * 3;
```

Comment:
We may later want to plan this out too, but the `* 3` is a hacky upper bound about how many fields one might write, and we could plan out the locations as well (e.g. `count(*)` needs only one, I think).

Reply:
im going to punt on this for now

Reply:
good!